### PR TITLE
feat: compress amd64 Linux kernel using zstd

### PR DESCRIPTION
**Pull request by smira**
This increases the kernel size, but moves us away from xz and improves decompression speed.

Also enable ZSTD firmware loading (not used yet).

Linux arm64 kernel is compressed with zstd already in 1.8+.

### CODE REVIEW

**Review by dsseng — APPROVED** [review]

File: kernel/build/config-amd64 [hunk 1]
```diff
@@ -16,6 +16,7 @@ CONFIG_CC_CAN_LINK_STATIC=y
 CONFIG_CC_HAS_ASM_GOTO_OUTPUT=y
 CONFIG_CC_HAS_ASM_GOTO_TIED_OUTPUT=y
 CONFIG_GCC_ASM_GOTO_OUTPUT_WORKAROUND=y
+CONFIG_TOOLS_SUPPORT_RELR=y
```
Comment (frezbo) [inline kernel/build/config-amd64:19]: this thing is a pain now, `menuconfig` removes them, `olddefconfig` adds it back :sweat_smile:

**Review by frezbo — COMMENTED** [review]
does that mean we have to compress firmware with zstd?

**Review by frezbo — APPROVED** [review]

**Comment by smira** [comment]
> does that mean we have to compress firmware with zstd?

I don't think we compress firmware at all, but it can load compressed if we decide to; I think having support doesn't hurt

**Comment by frezbo** [comment]
> > does that mean we have to compress firmware with zstd?
> 
> I don't think we compress firmware at all, but it can load compressed if we decide to; I think having support doesn't hurt

cool

**Comment by dsseng** [comment]
> > does that mean we have to compress firmware with zstd?
> 
> I don't think we compress firmware at all, but it can load compressed if we decide to; I think having support doesn't hurt

I can try looking into firmware compression someday. Perhaps we should open an issue for that to be looked into. Also we might want to consider further reducing firmware size by cutting out wireless and other firmwares we don't actually use because drivers aren't enabled

**Pull request merged**